appcues/src/main/java/com/appcues/analytics/Analytics.kt

```diff
@@ -0,0 +1,236 @@
+}
+
+internal interface RenderingService {
+
+    // abstraction about who is responsible for tracking experience events
+    interface EventTracker {
+
+        fun trackEvent(name: String, properties: Map<String, Any>?, isInteractive: Boolean, isInternal: Boolean)
+    }
+
+    fun setEventTracker(eventTracker: EventTracker)
+
+    // will attempt to show experiences by order
+    // clearCache is mostly used when we know the qualification is from a screen_view
+    suspend fun show(experiences: List<Experience>, clearCache: Boolean)
+
+    suspend fun reset()
+}
+
+internal interface SessionService {
+
+    // checks session or creates one based on existing information or incoming intent
+    suspend fun checkSession(intent: AnalyticsIntent, onSessionStarted: suspend () -> Unit): Boolean
+
+    // produce session properties
+    suspend fun getSessionProperties(): SessionProperties?
+
+    suspend fun reset()
+}
+
+internal data class SessionProperties(
+    val sessionId: UUID,
+    val userId: String,
+    val groupId: String?,
+    val userSignature: String?,
+    val latestUserProperties: Map<String, Any>,
+    val properties: Map<String, Any>,
+)
+
+internal interface ActivityBuilder {
+
+    suspend fun buildActivity(intent: AnalyticsIntent, sessionProperties: SessionProperties?): AnalyticsActivity?
+}
+
+// new model for Intents available for Analytics
+internal sealed class AnalyticsIntent(open val isInternal: Boolean, open val properties: Map<String, Any>?, val timestamp: Date = Date()) {
+
+    object Anonymous : AnalyticsIntent(false, null)
+
+    data class Identify(
+        val userId: String,
+        override val properties: Map<String, Any>?
+    ) : AnalyticsIntent(false, properties)
+
+    data class UpdateProfile(
+        override val isInternal: Boolean,
+        override val properties: Map<String, Any>?
+    ) : AnalyticsIntent(isInternal, properties)
+
+    data class UpdateGroup(
+        override val isInternal: Boolean,
+        val groupId: String?,
+        override val properties: Map<String, Any>?
+    ) : AnalyticsIntent(isInternal, properties)
+
+    data class Event(
+        override val isInternal: Boolean,
+        val name: String,
+        override val properties: Map<String, Any>?
+    ) : AnalyticsIntent(isInternal, properties)
+
+    data class Screen(
+        override val isInternal: Boolean,
+        val title: String,
+        override val properties: Map<String, Any>?
+    ) : AnalyticsIntent(isInternal, properties)
+}
+
+internal data class AnalyticsActivity(
+    val type: AnalyticType,
+    val isInternal: Boolean,
+    val timestamp: Date,
+    val sessionId: UUID,
+    val userId: String,
+    val groupId: String?,
+    val profileProperties: Map<String, Any>?,
+    val groupProperties: Map<String, Any>?,
+    val eventName: String?,
+    val eventAttributes: Map<String, Any>?,
+    val eventContext: Map<String, Any>?,
+    val userSignature: String?,
+)
+
+internal class Analytics(
```

Comment:
note about interfaces used here.

they are architectural bounds, we may them or not in the future, but the idea is that this "core" piece of our project could be isolated in a module, and we could have different modules providing the implementation on those dependencies.